eodatasets3/scripts/tostac.py

```diff
@@ -7,70 +7,182 @@
+    :return: Converted value with required type in STAC
+
+    """
+    # In STAC spec, "instruments" have [String] type
+    if key == "eo:instrument":
+        return [value]
+    else:
+        return value
+
+
 @click.command(help=__doc__)
+@click.option("--stac-template", "-t", type=str)
+@click.option("--stac-base-url", "-u", default="", type=str)
+@click.option(
+    "--product-base-url", "-p", default="https://explorer.dea.ga.gov.au", type=str
+)
 @click.argument(
     "odc_metadata_files",
     type=PathPath(exists=True, readable=True, writable=False),
     nargs=-1,
 )
-def run(odc_metadata_files: Iterable[Path]):
+def run(
+    odc_metadata_files: Iterable[Path], stac_template, stac_base_url, product_base_url
+):
     for input_metadata in odc_metadata_files:
         dataset = serialise.from_path(input_metadata)
 
-        project = partial(
-            pyproj.transform,
-            pyproj.Proj(init=dataset.crs),
-            pyproj.Proj(init="epsg:4326"),
-        )
-        wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
-        item_doc = dict(
-            id=dataset.id,
-            type="Feature",
-            bbox=wgs84_geometry.bounds,
-            geometry=wgs84_geometry.__geo_interface__,
-            properties={**dataset.properties, "odc:product": dataset.product.name},
-            assets={
-                # TODO: Currently assuming no name collisions.
-                **{name: {"href": m.path} for name, m in dataset.measurements.items()},
-                **{name: {"href": m.path} for name, m in dataset.accessories.items()},
-            },
-            links=[
-                # {
-                #     "rel": "self",
-                #     "href": '?',
-                # },
-                {"rel": "odc_product", "href": dataset.product.href},
-                {
-                    "rel": "alternative",
-                    "type": "text/html",
-                    "href": f"https://explorer.dea.ga.gov.au/dataset/{dataset.id}",
-                },
-            ],
-        )
-
         name = input_metadata.stem.replace(".odc-metadata", "")
         output_path = input_metadata.with_name(f"{name}.stac-item.json")
 
+        # Load STAC template data
+        if stac_template:
+            with open(stac_template, "r") as f:
+                stac_data = json.load(f)
+        else:
+            stac_data = {}
+
+        # Create STAC dict
+        item_doc = create_stac(
+            dataset,
+            input_metadata,
+            output_path,
+            stac_data,
+            stac_base_url,
+            product_base_url,
+        )
+
         with output_path.open("w") as f:
             json.dump(item_doc, f, indent=4, default=json_fallback)
 
-        echo(output_path)
+
+def create_stac(
+    dataset, input_metadata, output_path, stac_data, stac_base_url, product_base_url
+):
+    """
+    Create STAC document
+
+    :param dataset: Dict of the metadata content
```

Comment:
Not a high priority, but in most of this repo we're using type annotations in the function parameters rather than describing the type in English (see the `run()` function above).

Reply:
Used type annotations for consistent code style